### ISSUE TITLE
False positive in FinalLocalVariable check on unnamed variables

### ISSUE THREAD

**Issue by mahfouz72** [checkstyle/checkstyle#14981]
child #14942
I have read check documentation: https://checkstyle.org/checks/coding/finallocalvariable.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```
PS D:\CS\test> cat src/Test.java                                                 
import java.util.PriorityQueue;
import java.util.Queue;

public class Test {
    public void testParenPad(Point p) {
        final Queue<Integer> q = new PriorityQueue<>();
        q.add(1);
        q.add(2);
        for (Integer i : q) {
            var _ = q.poll();  // violation
        }
    }
}
PS D:\CS\test> cat config.xml                                                    
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>
    <module name="TreeWalker">
        <module name="FinalLocalVariable"/>
    </module>
</module>
PS D:\CS\test> java  -jar checkstyle-10.17.0-all.jar -c config.xml  src/Test.java
Starting audit...
[ERROR] D:\CS\test\src\Test.java:10:17: Variable '_' should be declared final. [FinalLocalVariable]
Audit done.
Checkstyle ends with 1 errors.
```

---

**Describe what you expect in detail.**
no violation this is an unnamed variable that is effectively final

---

**Comment by nrmancuso** [comment]
I am good to approve if we add a new property, `validateUnnamedVariables`. @rnveach please share your thoughts and approve if you agree.

**Comment by rnveach** [comment]
> validateUnnamedVariables

[We have 6 "validate" properties.  We have 46 "allow" properties. We have 22 "violate" properties. We have 48 "ignore" properties.](https://rveach.no-ip.org/checkstyle/propertyList.txt)

I was going to suggest "allow", maybe "ignore", but I don't know if it really matters. I guess part of my reason is I like simpler words, but also this seems more like a allow/ignore situation (maybe more ignore). The statistics seemed fun to look at. I am good with what you decide.

**Comment by nrmancuso** [comment]
I went with `validate` since the other check property is `validateEnhancedForLoopVariable` to be consistent.
